components/script/dom/dedicatedworkerglobalscope.rs

```diff
@@ -344,7 +345,16 @@ impl DedicatedWorkerGlobalScope {
                     .referrer_policy(referrer_policy)
                     .origin(origin);
 
-                let runtime = unsafe { new_child_runtime(parent) };
+                let runtime = unsafe {
+                    if let Some(pipeline_id) = pipeline_id {
```

Comment:
It looks like the pipeline_id in `WorkerScriptLoadOrigin` is always some, so you might want to address it in this PR as well by removing the option.

Reply:
Filed a followup: https://github.com/servo/servo/issues/24772